### ISSUE TITLE
Override TR_ResolvedJ9Method::isCompilable() for JITServer

### ISSUE THREAD

**Issue by mpirvu** [eclipse-openj9/openj9#17931]
`TR_ResolvedJ9Method::isCompilable()` accesses some fields from JavaVM and it's not safe to use as is in out-of-process compilations.
To fix this we have two alternatives:

1. When the server constructs a ResolvedJ9Method, a mirror resolved method is also built at the client. When that happens we can precompute the `isCompilable` information and store it into a boolean field which will be passed to the server. The disadvantage is that we have to precompute this field even though the optimizer may not ask about it.
2. For each client, store at the server the JavaVM fields that `isCompilable` is asking for.

Solution 2 should be preferred because it has less overhead. However, solution 2 is contingent on these JavaVM fields being populated exactly once before first compilation takes place:
javaVM->doPrivilegedWithContextMethodID1
javaVM->doPrivilegedWithContextMethodID2
javaVM->doPrivilegedWithContextPermissionMethodID1
javaVM->doPrivilegedWithContextPermissionMethodID2
javaVM->nativeLibrariesLoadMethodID

**Comment by mpirvu** [comment]
I see code that tests whether `doPrivilegedWithContextMethodID1` is NULL. So, this field starts as NULL and when the Java program executes the JNI call java/security/AccessController.initializeInternal(), that field (and others) are populated.
Interestingly I found on the internet that `java.security.AccessController` is deprecated for removal since Java17

**Comment by mpirvu** [comment]
Another field that ResolveJ9Method looks at is  `javaVM->nativeLibrariesLoadMethodID;`
This is set in `standardInit(J9JavaVM *vm, char *dllName)`, but only for Java >= 15
```
if JAVA_SPEC_VERSION >= 15
				JNIEnv *env = (JNIEnv*)vmThread;
				jclass clz = (*env)->FindClass(env, "jdk/internal/loader/NativeLibraries");
				jmethodID mid = NULL;
				if (NULL == clz) {
					goto _fail;
				}
#if (17 <= JAVA_SPEC_VERSION) && (JAVA_SPEC_VERSION <= 18)
				mid = (*env)->GetStaticMethodID(env, clz, "load", "(Ljdk/internal/loader/NativeLibraries$NativeLibraryImpl;Ljava/lang/String;ZZZ)Z");
#else /* (17 <= JAVA_SPEC_VERSION) && (JAVA_SPEC_VERSION <= 18) */
				mid = (*env)->GetStaticMethodID(env, clz, "load", "(Ljdk/internal/loader/NativeLibraries$NativeLibraryImpl;Ljava/lang/String;ZZ)Z");
#endif /* (17 <= JAVA_SPEC_VERSION) && (JAVA_SPEC_VERSION <= 18) */
				if (NULL == mid) {
					goto _fail;
				}
				vm->nativeLibrariesLoadMethodID = (UDATA)mid;
				Trc_JCL_init_nativeLibrariesLoadMethodID(vmThread, vm->nativeLibrariesLoadMethodID);
#endif /* JAVA_SPEC_VERSION >= 15 */
```
The method id corresponds to  `jdk/internal/loader/NativeLibraries.load(Ljdk/internal/loader/NativeLibraries$NativeLibraryImpl;Ljava/lang/String;ZZZ)Z`

**Comment by mpirvu** [comment]
Interesting point:
`TR_ResolvedJ9MethodBase::isCompilable(TR_Memory * trMemory)` already rejects the following methods (based on name)
```
"java/security/AccessController.doPrivileged(Ljava/security/PrivilegedAction;Ljava/security/AccessControlContext;)Ljava/lang/Object;",
"java/security/AccessController.doPrivileged(Ljava/security/PrivilegedExceptionAction;Ljava/security/AccessControlContext;)Ljava/lang/Object;",
"java/security/AccessController.doPrivileged(Ljava/security/PrivilegedAction;Ljava/security/AccessControlContext;[Ljava/security/Permission;)Ljava/lang/Object;",
"java/security/AccessController.doPrivileged(Ljava/security/PrivilegedExceptionAction;Ljava/security/AccessControlContext;[Ljava/security/Permission;)Ljava/lang/Object;",
```

`TR_ResolvedJ9Method::isCompilable(TR_Memory * trMemory)` first applies the filters from `TR_ResolvedJ9MethodBase::isCompilable(trMemory)` and filters out
`javaVM->doPrivilegedWithContextMethodID1`
`javaVM->doPrivilegedWithContextMethodID2`
`javaVM->doPrivilegedWithContextPermissionMethodID1;`
`javaVM->doPrivilegedWithContextPermissionMethodID2;`
which correspond to
`doPrivileged(Ljava/security/PrivilegedAction;Ljava/security/AccessControlContext;)Ljava/lang/Object;)`
`doPrivileged(Ljava/security/PrivilegedExceptionAction;Ljava/security/AccessControlContext;)Ljava/lang/Object;)`
`"doPrivileged(Ljava/security/PrivilegedAction;Ljava/security/AccessControlContext;[Ljava/security/Permission;)Ljava/lang/Object;)`
`"doPrivileged(Ljava/security/PrivilegedExceptionAction;Ljava/security/AccessControlContext;[Ljava/security/Permission;)Ljava/lang/Object;)`

There are also these two which are not captured by `TR_ResolvedJ9Method::isCompilable`
`doPrivileged(Ljava/security/PrivilegedAction;)Ljava/lang/Object;)`
`doPrivileged(Ljava/security/PrivilegedExceptionAction;)Ljava/lang/Object;)`

So, to recap:
1. We don't need the "doPrivileged" code in `TR_ResolvedJ9Method::isCompilable` because `TR_ResolvedJ9MethodBase::isCompilable()` handles those cases already.
2. We may need to add two extra `doPrivileged` methods in `TR_ResolvedJ9MethodBase::isCompilable()`

**Comment by mpirvu** [comment]
Even "jdk/internal/loader/NativeLibraries.load()" is handled by `TR_ResolvedJ9MethodBase::isCompilable(TR_Memory * trMemory)`, so the code can be cleaned-up a lot.

Proposed solution is to simplify `TR_ResolvedJ9Method::isCompilable(TR_Memory * trMemory)` by eliminating the following code
```
   /* AOT handles this by having the names in the string compare list */
   J9JavaVM * javaVM = _fe->_jitConfig->javaVM;
   J9JNIMethodID * magic = (J9JNIMethodID *) javaVM->doPrivilegedWithContextMethodID1;
   if (magic && ramMethod() == magic->method)
      return false;

   magic = (J9JNIMethodID *) javaVM->doPrivilegedWithContextMethodID2;
   if (magic && ramMethod() == magic->method)
      return false;

   magic = (J9JNIMethodID *) javaVM->doPrivilegedWithContextPermissionMethodID1;
   if (magic && ramMethod() == magic->method)
      return false;

   magic = (J9JNIMethodID *) javaVM->doPrivilegedWithContextPermissionMethodID2;
   if (magic && ramMethod() == magic->method)
      return false;

   magic = (J9JNIMethodID *) javaVM->nativeLibrariesLoadMethodID;
   if (magic && ramMethod() == magic->method) {
      return false;
   }
```
